### ISSUE TITLE
[Feature Request]: Faciliate "single-package" use

### ISSUE THREAD

**Issue by Biont** [inpsyde/wp-translation-downloader#25]
### Is your feature request related to a problem?

This wonderful composer plugin has been written for orchestrating entire WordPress installations. However, It would be very helpful for plugin development if we were able to include this tool into our development and packaging workflows. By this I mean the ability to download translations inside the repository of a single `"wordpress-plugin"` into its own `languages/`-Folder

When trying to integrate it, I was met with a few obstacles:

1. The plugin creates a sub-directory structure under the specified `extra.wp-translation-downloader.directory` (*library, plugins, themes*)
2. Only dependencies of the root package are processed, not the root package itself
3. Even when forced to process the root package by editing the code, the translations are consequently put into the newly created *languages/plugins/* folder instead of the root `languages/`

I am of course well aware *why* the plugin behaves the way it does and if I'm doing something that goes against the fundamental design of it, then I must find other solutions. But if would be pretty cool if it could work.

### Describe the desired solution

It seems to me that there's three things to deal with:

#### Take the root package into account in `DownloadCommand`

```php
// \Inpsyde\WpTranslationDownloader\Command\DownloadCommand::resolvePackages
$packages[]=$composer->getPackage();
```

#### Lazy creation of subdirectories

The *library, plugins, themes* directories are always created even if  nothing will be put in them. Creating them lazily would prevent polluting the filesystem with empty folders

#### Add a mechanism to control the target directory per-package
There's no way to make the translations of a `wordpress-plugin` go straight into `languages/`. A mechanism to configure the (sub-)directory handling could provide me with the tools needed.
As a suggestion, the `extra.wp-translation-downloader.directory` setting could optionally be an object that is structured similar to `extra.installer-paths`:
```json
{
  "languages/foo": ["type:wordpress-plugin"],
  "languages/bar": ["type:wordpress-theme"],
}
```

For my use case, I could basically just pass
```json
{
  "languages": ["type:wordpress-plugin"]
}
```

But these are just suggestions of course

### Describe the alternatives that you have considered

A complementary solution could be to add a command/flag that explicitly only downloads a single package's translations. This might have use-cases outside of my specific one as well (like updating translations of `PluginA` when you know the translations of `PluginB` are dirty for some reason, so a re-download of everything is not an option)

### Additional context

As an entertaining workaround, I can get pretty close to my desired outcome by declaring the root package to be of `"type": "wordpress-core"`

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Chrico** [comment]
Thanks for the input @Biont ! Totally understand your use case and yes, it might makes sense to include wp-translation-downloader into a plugin/library itself and install the root dependencies.

I was also considering to improve the "resolving of directories" due issue #13 to not restrict to specific types (library, wp core, plugin, theme) by static mapping those to folders like it is currently done in [`Config\PluginConfiguration`](https://github.com/inpsyde/wp-translation-downloader/blob/fd4a2c810f79157a9f25213e5e04a478cf15b64a/src/Config/PluginConfiguration.php#L68-L73). 
Instead it should be more flexible by allowing configuration and provding just _default_ locations which can be overwritten. It could be similar done in terms of configuration for "names" and "types" as it is for the "api".

```json
{
	"languageRootDir": "public/wp-content/languages",
	"directories": {
		"names": {
			"inpsyde/some-library": "different-folder/"
		}
		"types": {
			"wordpress-theme": "themes/"
		}
	}
}
```

---

I guess a good step would be to add the root package, which would solve the use case:

> i want to execute wp-translation-downloader in my Plugin which is the root package.

**Comment by Chrico** [comment]
Started to work here: https://github.com/inpsyde/wp-translation-downloader/tree/feature-custom-directories @Biont

**Comment by Chrico** [comment]
Merged today the branch via https://github.com/inpsyde/wp-translation-downloader/pull/27 into `dev-master`. No release created yet, but will come soon. ;-)

**Comment by Chrico** [comment]
This was released with the latest 2.1 version. Check out the changelog: https://github.com/inpsyde/wp-translation-downloader/releases/tag/2.1 :)

**Comment by Biont** [comment]
Thank you very much for this!
Since this new release appears to cover everything that was raised in this issue, I think we can close it. Remaining problems can be discussed in new issues. Thanks again.
